File: 프로그래머스/1/42840. 모의고사/모의고사.java
import java.util.*;
class Solution {
    public int[] solution(int[] answers) {
        int[] answer = {};
        int[][] people = {{1, 2, 3, 4, 5},
                          {2, 1, 2, 3, 2, 4, 2, 5},
                          {3, 3, 1, 1, 2, 2, 4, 4, 5, 5}
                         };
        int[] point = {0, 0, 0};
        
        int n = answers.length;
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < 3; j++) {
                int m = people[j].length;
                if (answers[i] == people[j][i % m]) point[j]++;
            }
        }
        int max = Math.max(point[0], Math.max(point[1], point[2]));
        
        ArrayList<Integer> list = new ArrayList<>();
        for (int i = 0; i < 3; i++) {
            if (point[i] == max) {
                list.add(i + 1);
            }
        }
        
        return list.stream().mapToInt(Integer::intValue).toArray();
    }
}

File: 프로그래머스/1/42840. 모의고사/모의고사.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> answers) {
    vector<int> answer;
    vector<int> su1 = {1, 2, 3, 4, 5};
    vector<int> su2 = {2, 1, 2, 3, 2, 4, 2, 5};
    vector<int> su3 = {3, 3, 1, 1, 2, 2, 4, 4, 5, 5};
    vector<vector<int>> su = {su1, su2, su3};
    
    vector<int> a(3);
    for(int j = 0; j < 3; j++) {
        for(int i = 0; i < answers.size(); i++) {
            if(answers[i] == su[j][i % su[j].size()]) {
                a[j]++;
            }
        }
    }
    
    int max_val = *max_element(a.begin(), a.end());
    for(int i = 0; i < a.size(); i++) {
        if (max_val == a[i]) {
             answer.push_back(i + 1);
        }
    }
    
    return answer;
}